### DISCUSSION TITLE
Identify clicked data point in Chart onClick events

### DISCUSSION THREAD

**Discussion by paulgrieselhuber** [TanStack/react-charts#237] [Q&A] [answered]
Using the latest v3 beta and testing and a review of all examples I could find, and searching this repo for `onClick`, I could not find a way to resolve which data point had been clicked from the data passed into `onClick`.

Having explored the `SyntheticBaseEvent`, nothing stands out to me as a reference to the actual data set.

We plan to use these references to render a view that "drills down" into the selected data point.

Is there something I am missing on this event, or do we need to calculate the data point based on the size of the view and the (x, y) coordinates of the event?

**Comment by Brodzko** [comment, marked as answer]
Where are you passing your `onClick` handler? I believe this is what `onClickDatum: (datum, event) => void` in `Chart` options  is for. See the Line chart in this [forked example](https://codesandbox.io/s/cocky-cartwright-qipif).

**Reply by tannerlinsley** [reply]
Using the `onClickDatum` callback should give you exactly what you are looking for.

**Reply by paulgrieselhuber** [reply]
Bingo, that was it. Thanks @Brodzko & @tannerlinsley.
